lib/memstore.ts

```diff
@@ -39,24 +39,30 @@ import {
 import { Store } from './store'
 import { getCustomInspectSymbol, getUtilInspect } from './utilHelper'
 
-export class MemoryCookieStore extends Store {
-  override synchronous: boolean
-  idx: {
-    [domain: string]: {
-      [path: string]: {
-        [key: string]: Cookie
-      }
+export type MemoryCookieStoreIndex = {
+  [domain: string]: {
+    [path: string]: {
+      [key: string]: Cookie
     }
   }
+}
+
+export class MemoryCookieStore extends Store {
+  override synchronous: boolean
+  idx: MemoryCookieStoreIndex
 
   constructor() {
     super()
     this.synchronous = true
-    this.idx = Object.create(null)
+    this.idx = Object.create(null) as MemoryCookieStoreIndex
```

Comment:
```suggestion
    // `Object.create` returns `any`, but we know it's safe to use here
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    this.idx = Object.create(null)
```
[nit] Since it's eslint that's complaining, I'd fix it at the eslint level, rather than using a type assertion.